superset/db_engine_specs/base.py

```diff
@@ -132,7 +138,9 @@ def get_allow_cost_estimate(cls, version: str = None) -> bool:
         return False
 
     @classmethod
-    def get_engine(cls, database, schema=None, source=None):
+    def get_engine(
+        cls, database: Database, schema: str = None, source: str = None
```

Comment:
If you've enabled `no implicit optional` shouldn't these be `Optional[str]`?

Reply:
Agreed. I'm currently unsure why Mypy isn't enforcing that per [here](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-no-implicit-optional).